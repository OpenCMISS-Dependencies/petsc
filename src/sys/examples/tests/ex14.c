
static char help[] = "Tests PetscOptionsGetScalar() for complex numbers\n";

#include <petscsys.h>

#undef __FUNCT__
#define __FUNCT__ "main"
int main(int argc,char **argv)
{
  int         ierr;
  PetscScalar a;

  PetscInitialize(&argc,&argv,(char*)0,help);
  ierr = PetscOptionsGetScalar(NULL,"-a",&a,NULL);CHKERRQ(ierr);
  ierr = PetscPrintf(PETSC_COMM_SELF,"Scalar a = %g + %gi\n",(double)PetscRealPart(a),(double)PetscImaginaryPart(a));CHKERRQ(ierr);
  ierr = PetscFinalize();
  return 0;
}

