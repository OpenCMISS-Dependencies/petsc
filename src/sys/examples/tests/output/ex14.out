Scalar a = 1.5 + 2.1i
Real rarray of length 3
 1,
 2,
 3,
Scalar rarray of length 8
 1 + 0i
 -2 - 3i
 4.5 + 6.2i
 4.5 + 0i
 6.8 + 4i
 0 + 1i
 -0 - 1i
 0 - 1.2i
